manifests/init.pp

```diff
@@ -46,44 +51,58 @@
 # Copyright 2013 Travis F, unless otherwise noted.
 #
 define pget (
-  $source,                 #: the source file location, supports local files, http://, https://, ftp://
-  $target         = undef, #: the target stage directory,
-  $targetfilename = undef, #: desired filename, will be infered by the source if not defined.
-  $username       = undef, #: Username to be passed
-  $password       = undef, #: password needed,
-  $timeout        = 300,   #: timeout
-  $headerHash     = undef, #: additional has parameters for the download of the file, i.e. user-agent, Cookie
-  ){
-
+  $source, # : the source file location, supports local files, http://, https://, ftp://
+  $target         = undef, # : the target stage directory,
+  $targetfilename = undef, # : desired filename, will be infered by the source if not defined.
+  $username       = undef, # : Username to be passed
+  $password       = undef, # : password needed,
+  $timeout        = 300,   # : timeout
+  $headerHash     = undef, # : additional has parameters for the download of the file, i.e. user-agent, Cookie
+  $overwrite      = false, # : if the target file should be overwritten (if already present)
+  ) {
   validate_string($source)
-  validate_re($source,['^s?ftp:','^https?:','^ftps?:','^puppet:'])
+  validate_re($source, [
+    '^s?ftp:',
+    '^https?:',
+    '^ftps?:',
+    '^puppet:'])
 
-  if $::operatingsystem != 'windows'{
+  if $::operatingsystem != 'windows' {
     fail("Unsupported OS ${::operatingsystem}")
   }
-  
+
   if $targetfilename != undef {
-     $filename = $targetfilename
+    $filename = $targetfilename
   } else {
-	   $filename = pget_filename($source)
+    $filename = pget_filename($source)
   }
-  
+
   $target_file = "${target}/${filename}"
 
+  if $overwrite {
+    $unlessClause = ''
```

Comment:
This instead should be undef and passed accordingly.  Otherwise it will still try to run the check.